### ISSUE TITLE
New 3.2.3 release actually reports as 2.3.1 in UI.

### ISSUE THREAD

**Issue by ThomasTrain** [Pro/dkim-exchange#289]
ExchangeDkimSigner.dll file is still reporting 2.3.1 version.

**Comment by DJBenson** [comment]
Can you tell me where you are seeing that version number reported as I've just checked the latest build on my Exchange 2019 CU4 box (installed just now from the v3.2.3 build I just published) and all the file versions are correctly reporting 3.2.3.

ExchangeDkimSigner.dll Windows Properties:
![Screenshot_1](https://user-images.githubusercontent.com/1013909/71910695-4dd01b80-316a-11ea-8405-a447fe3cb6b0.png)

ExchangeDkimSigner.exe 
![Screenshot_4](https://user-images.githubusercontent.com/1013909/71910898-ab646800-316a-11ea-9802-f61c24832351.png)
Windows Properties:

DKIM Signer Information Tab:
![Screenshot_2](https://user-images.githubusercontent.com/1013909/71910701-50327580-316a-11ea-95d8-10c13801fd1c.png)

DKIM Signer About Tab:
![Screenshot_3](https://user-images.githubusercontent.com/1013909/71910703-5163a280-316a-11ea-9313-384d49d2bf6d.png)

Can you try reinstalling the application (or post a screenshot of where that version number is coming from).

**Comment by ThomasTrain** [comment]
OK, so I was originally running Exchange 2013 CU13 anddid an upgrade of DKIM exchange using the UI upgrade option.  All of the binaries were upgraded with new timestamps but the version number shown in the UI and on file details for ExchangeDkimSigner.dll remained as 3.2.1.

Since I needed the DKIM update to upgrade to CU15 I went ahead and completed that upgrade. 
 After completing the upgrade from CU13 to CU15 and upon conclusion reinstalling DKIM Exchange (via the UI since it was sitll showing as 3.2.1) and now the version in the UI and on details for ExchangeDkimSigner.dll is now showing at 3.2.3.

That said, after inspection of the source, the ExchangeDkimSigner.dll doens't appear to have been properly recompiled for all of the exchange versions.  I noticed that under dkim-exchange-3.2.3\Src\Exchange.DkimSigner\bin Exchange 2016 CU10 - CU13 were still using 3.2.1 version of the dll file.  Exchange 2016 CU8 - CU9 are using 3.2.0. etc.

I didn't do any more checks but it does look like an issue with the compiled release version of ExchangeDkimSigner.dll across a number of Exchange versions.

**Comment by ThomasTrain** [comment]
Big thank you for jumping in and fixing that so fast!

**Comment by DJBenson** [comment]
> Big thank you for jumping in and fixing that so fast!

No problem. I've published the fix as a pre-release for now. If there are no more problems with it, I'll merge it into master at some point before the next release.

Thanks for raising this, it was an odd one, but probably down to missing dependencies on my build machine - why VS was not throwing an error I'm not sure but it seemed to silently skip a hand ful of DLLs but then failed when I manually tried to compile the missing ones (which ultimately led me to the missing dependencies issue).
